geomstats/geometry/product_manifold.py

```diff
@@ -14,101 +14,288 @@
-
-    def __init__(
-        self, manifolds, metrics=None, default_point_type="vector", n_jobs=1, **kwargs
-    ):
+    def __init__(self, factors, metrics=None, default_point_type="vector", **kwargs):
         geomstats.errors.check_parameter_accepted_values(
             default_point_type, "default_point_type", ["vector", "matrix"]
         )
 
-        self.dims = [manifold.dim for manifold in manifolds]
+        self.factors = factors
+
+        dim = sum(self.factor_dims)
+
+        if default_point_type == "vector":
+            shape = (sum([prod(factor_shape) for factor_shape in self.factor_shapes]),)
+        else:
+            if (self.factor_shapes.count(self.factor_shapes[0]) ==
+                    len(self.factor_shapes)):
+                if len(self.factor_shapes[0]) == 1:
+                    shape = (len(self.factors), *self.factors[0].shape)
+                else:
+                    raise ValueError(
+                        "A default_point_type of \'matrix\' can only be used if all "
+                        "manifolds have vector type."
+                    )
+            else:
+                raise ValueError(
+                    "A default_point_type of \'matrix\' can only be used if all "
+                    "manifolds have the same shape."
+                )
+
+        if "extrinsic" in self.factor_default_coords_types:
+            default_coords_type = "extrinsic"
+        else:
+            default_coords_type = "intrinsic"
+
         if metrics is None:
-            metrics = [manifold.metric for manifold in manifolds]
+            metric_scales = None
+            metrics = [manifold.metric for manifold in factors]
+        elif gs.all([is_positive(metric) for metric in metrics]):
+            metric_scales = metrics
+            metrics = [manifold.metric for manifold in factors]
+        else:
+            metric_scales = None
+
         kwargs.setdefault(
             "metric",
             ProductRiemannianMetric(
-                metrics, n_jobs=n_jobs, default_point_type=default_point_type
-            ),
+                metrics, default_point_type=default_point_type, scales=metric_scales),
         )
-        dim = sum(self.dims)
-
-        if default_point_type == "vector":
-            shape = (sum([m.shape[0] for m in manifolds]),)
-        else:
-            shape = (len(manifolds), *manifolds[0].shape)
 
         super().__init__(
             dim=dim,
             shape=shape,
+            default_coords_type=default_coords_type,
             **kwargs,
         )
-        self.manifolds = manifolds
-        self.n_jobs = n_jobs
+
+        if self.default_coords_type == "extrinsic":
+            factor_embedding_spaces = [
+                manifold.embedding_space if manifold.default_coords_type == "extrinsic"
+                else manifold
+                for manifold in factors
+            ]
+            self.embedding_space = ProductManifold(factor_embedding_spaces)
```

Comment:
the `scale` of factors has no importance for the scales of the embedding spaces?